### PR TITLE
ceph.in: ability to change file ownership

### PR DESCRIPTION
**Pull request by leseb**
When creating/fetching key it's nice to have the ability to change the
ownership of the created file.

This commit adds the '--setuser' and 'setgroup' which respectively apply
the desired owner and group to a file user when '--output' is passed.

Closes: https://tracker.ceph.com/issues/38370
Signed-off-by: Sébastien Han <seb@redhat.com>

- [x] References tracker ticket
- [x] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by leseb** [comment]
I didn't find any tests, happy to add some if you think it's necessary.

File: src/ceph.in [hunk 1]
```diff
@@ -1107,6 +1112,20 @@ def main():
         except Exception as e:
             print('Can\'t open output file {0}: {1}'.format(parsed_args.output_file, e), file=sys.stderr)
             return 1
+        if parsed_args.setuser:
+            try:
+                owner = pwd.getpwnam(parsed_args.setuser).pw_uid
+                os.chown(parsed_args.output_file, owner, -1)
```
Comment (batrick) [inline src/ceph.in]: `fchown` would be preferred

File: src/ceph.in [hunk 2]
```diff
@@ -1107,6 +1112,20 @@ def main():
...
+            except OSError as e:
+                print('Failed to change user ownership of {0} to {1}: {3}'.format(outf, parsed_args.setuser, e))
+                return 1
+        if parsed_args.setgroup:
+            try:
+                group = grp.getgrnam(parsed_args.setgroup).gr_gid
+                os.chown(parsed_args.output_file, -1, group)
```
Comment (batrick) [inline src/ceph.in]: ditto

**Comment by leseb** [comment]
@batrick done.

File: src/ceph.in [hunk 3]
```diff
@@ -1107,6 +1112,22 @@ def main():
         except Exception as e:
             print('Can\'t open output file {0}: {1}'.format(parsed_args.output_file, e), file=sys.stderr)
             return 1
+        if parsed_args.setuser:
+            try:
+                ownerid = pwd.getpwnam(parsed_args.setuser).pw_uid
+                with open(parsed_args.output_file, 'wb') as f:
+                    os.fchown(f.fileno(), ownerid, -1)
```
Comment (batrick) [inline src/ceph.in]: um, why not `outf.fileno()`?

**Comment by leseb** [comment]
Anything else I should do to see this merged? Thanks

**Comment by batrick** [comment]
@liewegas can you add this to your next RADOS QA run

**Comment by batrick** [comment]
@leseb do you want this backported to luminous/mimic? please update the tracker ticket.

**Comment by leseb** [comment]
@batrick would be nice yes, ok done.

**Comment by leseb** [comment]
Are we good to go? This drastically helps https://github.com/rook/rook/issues/2664.
Thanks!

**Comment by liewegas** [comment]
running it through the rados suite now

**Pull request merged**